File: downloads/python/labs2/ISU/Jewels/Icon.py
#!/usr/bin/env python
# coding: UTF-8
#
## @package Icon
#
#  Icon Interface.
#  Interface representing an icon or block in a Bejeweled-like matching game.
#  At a minimum, each icon encapsulates an integer, referred to as its "type".
#
#  @author Paulo Roma
#  @since 05/04/2020
#
try:                 # python >= 3.4
    from abc import ABC, ABCMeta, abstractmethod
except ImportError:  # python 2
    from abc import ABCMeta, abstractmethod
    ABC = object

##
# Interface representing an icon or block in a
# Bejeweled-like matching game. At a minimum, each
# icon encapsulates an integer, referred to as its "type".
#
class Icon(ABC):
    ##
    # Returns the type of this icon.
    # @return
    #   type of this icon
    #
    @abstractmethod
    def getType(self): pass

File: 3-javascript/jewels/jewels/Icon.js
/** @file
 *
 *  Summary.
 *  <p>Icon Interface.</p>
 *
 *  Description.
 *  <p>Interface representing an icon or block in a Bejeweled-like matching game. </p<
 *  <p> At a minimum, each icon encapsulates an integer, referred to as its "type". </p>
 *
 *  @author Paulo Roma
 *  @see <a href="/cwdc/3-javascript/jewels/Icon.js">source</a>
 *  @since 10/03/2021
 */

/**
 * Interface representing an icon or block in a Bejeweled-like matching game.<br>
 * At a minimum, each icon encapsulates an integer, referred to as its "type".
 * @interface
 */
export class Icon {
  constructor() {
    if (this.constructor === Icon) {
      throw new TypeError(
        'Abstract class "Icon" cannot be instantiated directly.'
      );
    }
  }

  /** Returns the type of this icon.
   *  @return {Number} type of this icon.
   */
  getType() {
    throw new Error("You have to implement the method getType!");
  }
}
